src/factors/Inertial/IMUDeltaFactor.jl

```diff
@@ -340,6 +326,9 @@ end
 IIF.getManifold(::IMUDeltaFactor) = IMUDeltaGroup()
 
 function IIF.preambleCache(fg::AbstractDFG, vars::AbstractVector{<:DFGVariable}, ::IMUDeltaFactor)
+    # FIXME, nsec should only contain fractional second information, i.e. < 1.0s.  Use together with `trunc(timestamp) + 1e-9*nsec`
```

Comment:
That is not how I use it and not how its documented.

Reply:
Again, we should do what ROS does and not our own special thing.  By the way, the suggested code here works for both cases.  We should also update our documentation to reflect the same as ROS.

Reply:
We had this discussion before and also didn't agree then.
If you want to do what ros does:
Split the timestamp in two 32 bit timestamps, one for your ns and the other for the second part. The 2 combined as a single 64bit timestamp is basically what you have here.

> By the way, the suggested code here works for both cases. We should also update our documentation to reflect the same as ROS.

I feel like a parrot, but it doesn't, I've had multiple cases where there are 2 different timestamps. Not all timestamps are synced, and not all timestamps are unix epoch.

Reply:
What we should not do is say `getTimestamp(var)` is not actually the timestamp.

Perhaps we should rename `.nstime` field to: `variable.missionnanosec` as part of DFG v1.  The suggested comments in code here already support both ("mission time" and ROS standard) cases where the full mission time is in the `nanosecond` field, or is split with the `timestamp` field.

This way the poweruser would know that `timestamp` and `missionnanosec` are equivalent.  They can then have different start times also, namely actual calendar `datetime+timezone`, and `0ns`.  They can then be made equivalent with 
`datetime2unix(timestamp[1]) + missionnanosec*1e-9`
but this sum is the really large number where Float64 risk of precision occurs.  That would be in line with `DateTime` being millisecond only.

We should obviously be using Float64 for this, so don't mind ROS' use of Float32.  Remember also that much of the web still uses 32 for many of its workings.

The suggested `missionnanosec` approach seems to solve all the problems?

Reply:
https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/issues/1087

Reply:
> 32 bit timestamps

I mean that we stay with 64 bit (obviously).

Note some web stacks only support 32 bit at this time.  Some of the json libraries was the example we bumped into before.

Reply:
updated the `FIXME` comment to be inline with `missionnanosec`.